### PR TITLE
watcher: Avoid using global default slog

### PR DESCRIPTION
**Pull request by sayboras**
As highlighted in below comment, the default logger is completely replaced (rather than being mutated) during the configuration phase, so we should avoid using global slog instance.

Relates: #35567

Reported-by: Marco Iorio <marco.iorio@isovalent.com>

### CODE REVIEW

File: operator/watchers/logfields.go
Comment (giorio94) [inline operator/watchers/logfields.go:1]: This file could be dropped altogether now.

**Review by rastislavs — APPROVED** [review]

File: operator/cmd/root.go [hunk 1]
```diff
@@ -588,7 +591,8 @@ func (legacy *legacyOnLeader) onStart(_ cell.HookContext) error {
 		if pooledAlloc, ok := alloc.(operatorWatchers.PooledAllocatorProvider); ok {
 			// The following operation will block until all pools are restored, thus it
 			// is safe to continue starting node allocation right after return.
-			operatorWatchers.StartIPPoolAllocator(legacy.ctx, legacy.clientset, pooledAlloc, legacy.resources.CiliumPodIPPools)
+			operatorWatchers.StartIPPoolAllocator(legacy.ctx, legacy.clientset, pooledAlloc, legacy.resources.CiliumPodIPPools,
+				legacy.logger.With(logfields.LogSubsys, "watchers"))
```
Comment (giorio94) [inline operator/cmd/root.go]: Nit. I'd suggest initializing the scoped logger only once, to ensure consistency and save a few allocations.

**Review by giorio94 — APPROVED** [review]
Thanks! I looks a bit weird having to pass the logger everywhere, but I agree it seems the least friction path for the moment, without requiring a large refactoring that is definitely out of scope here.

**Review by pippolo84 — APPROVED** [review]
LGTM minus Marco's comments :rocket:

**Comment by pippolo84** [comment]
Manually removing ready-to-merge label to give @sayboras a chance to address Marco's comments before merging.

**Comment by sayboras** [comment]
> Manually removing ready-to-merge label to give @sayboras a chance to address Marco's comments before merging.

Thanks a lot :+1:

**Pull request merged**